Description: Don't link QtWebkit with --gc-sections on powerpc and s390
 This works around a binutils bug that results in a segfault on powerpc
 and an internal link error on s390.
Author: Steve Langasek <steve.langasek@ubuntu.com>
Author: Modestas Vainius <modax@debian.org>
Origin: vendor
Forwarded: no
Last-Update: 2010-04-11 (s390 added)

--- a/src/3rdparty/webkit/WebCore/WebCore.pro
+++ b/src/3rdparty/webkit/WebCore/WebCore.pro
@@ -90,7 +90,7 @@ unix {
 }
 
 unix:!mac:*-g++*:QMAKE_CXXFLAGS += -ffunction-sections -fdata-sections 
-unix:!mac:*-g++*:QMAKE_LFLAGS += -Wl,--gc-sections
+unix:!mac:*-g++*:!equals(QT_ARCH, powerpc):!equals(QT_ARCH, s390):QMAKE_LFLAGS += -Wl,--gc-sections
 linux*-g++*:QMAKE_LFLAGS += $$QMAKE_LFLAGS_NOUNDEF
 
 CONFIG(release):!CONFIG(standalone_package) {
